perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/SecurityTeamsManagerBlImpl.java

```diff
@@ -111,19 +111,23 @@ public void deleteSecurityTeam(PerunSession sess, SecurityTeam securityTeam, boo
 		if (blacklist != null && !blacklist.isEmpty() && !forceDelete) {
 			throw new RelationExistsException("SecurityTeam has blacklisted users.");
 		}
-		for (User blacklistedUser : blacklist) {
-			// calling BL will make auditer message about user to appear.
-			getPerunBl().getSecurityTeamsManagerBl().removeUserFromBlacklist(sess, securityTeam, blacklistedUser);
+		if (blacklist != null) {
+			for (User blacklistedUser : blacklist) {
+				// calling BL will make auditer message about user to appear.
+				getPerunBl().getSecurityTeamsManagerBl().removeUserFromBlacklist(sess, securityTeam, blacklistedUser);
+			}
 		}
 
 		// remove security team from all facilities
 		List<Facility> facilities = getPerunBl().getFacilitiesManagerBl().getAssignedFacilities(sess, securityTeam);
 		if (facilities != null && !facilities.isEmpty() && !forceDelete) {
 			throw new RelationExistsException("SecurityTeam is assigned to some facilities.");
 		}
-		for (Facility facility : facilities) {
-			// calling BL will make auditer message about facility to appear.
-			getPerunBl().getFacilitiesManagerBl().removeSecurityTeam(sess, facility, securityTeam);
+		if (facilities != null) {
```

Comment:
Same here. This check is redundant. You should remove the check `facilities != null` from the condition above.